### ISSUE TITLE
Expose Fit Internals

### ISSUE THREAD

**Issue by curran** [d3/d3-geo#217]
Faced with the task of adding `fit` methods to [geo-albers-usa-territories](https://github.com/stamen/geo-albers-usa-territories) , I'm a bit stuck.

Ideally I could follow the pattern found in [albersUsa.js](https://github.com/d3/d3-geo/blob/master/src/projection/albersUsa.js), namely

```js
import {fitExtent, fitSize, fitWidth, fitHeight} from "./fit.js";
```

However, since the projection is not inside this repository, and those internals from `fit.js` are not exported from the top level of this package, I see no way to access them. Am I missing an approach that would work?

FWIW I did try 

```js
import { fitExtent, fitSize, fitWidth, fitHeight } from 'd3-geo/src/fit.js';
```

But that feels janky, and caused some headaches with Rollup (here's a [WIP PR with the attempt, for reference]()).

I propose to export these methods from the `d3-geo` package, for use by projections defined outside this project.

Thoughts?

**Comment by curran** [comment]
That might look something like adding the following to [index.js](https://github.com/d3/d3-geo/blob/master/src/index.js):

```js
export {
  fitExtent as geoProjectionFitExtent,
  fitSize as geoProjectionFitSize,
  fitWidth as geoProjectionFitWidth,
  fitHeight as geoProjectionFitHeight
} from "./projection/fit.js";
```

**Comment by Fil** [comment]
An alternative approach would be to finally offer a composite projections construct, that would deal with all the peculiarities (clipping, inverse, etc).

**Comment by curran** [comment]
Interesting idea! Were there already discussions along the lines of a composite projections construct?

**Comment by curran** [comment]
Considering submitting a PR with the change to expose these.

**Comment by Fil** [comment]
It looks like @harrystevens has an alternative here https://observablehq.com/@washpostgraphics/geo-albers-usa-pr

**Comment by HarryStevens** [comment]
> It looks like @HarryStevens has an alternative here https://observablehq.com/@washpostgraphics/geo-albers-usa-pr

Yep, I just kept copy-pasting code from d3-geo until it stopped throwing errors.

**Comment by nthitz** [comment]
any chance of publishing to npm for easier use off of observable?
